server/src/main/java/com/testsigma/service/TestStepService.java

```diff
@@ -574,8 +577,40 @@ private List<ActionTestDataMap> getMapsList(){
 
         return actionsMap;
     }
+    public void bulkDelete(Long[] ids) throws ResourceNotFoundException {
+        for (Long id:ids){
+              this.destroy(this.find(id));
+            }
+    }
 
-
+    private Boolean checkBulkDeleteStepsHaveAllPrerequisiteSteps(List<Long> ids,List<TestStep> testStepList){
+        for (TestStep testStep:testStepList){
+            if(!ids.contains(testStep.getId())){
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public List<TestStepDTO> checkForLinkedTestCases(Long[] ids) throws ResourceNotFoundException {
+        Long testCaseId = this.find(ids[0]).getTestCaseId();
+        List<TestStepDTO> testSteps = new ArrayList<>();
+        List<Long> idsList=Arrays.asList(ids);
+        int index=0;
+        for(Long id:ids) {
+            List<TestStep> testStepList= this.repository.findAllByTestCaseIdAndPreRequisiteStepId(testCaseId,id);
+            if(!testStepList.isEmpty() && !checkBulkDeleteStepsHaveAllPrerequisiteSteps(idsList,testStepList)){
+                TestStep mainStep = this.find(id);
+                TestStepDTO testStepDTO = this.exportTestStepMapper.mapDTO(mainStep);
+                testStepDTO.setPreRequisiteStepId(null);
+                testSteps.add(index++,testStepDTO);
+                for(TestStep testStep:testStepList){
+                    testSteps.add(index++,this.exportTestStepMapper.mapDTO(testStep));
+                }
+            }
+        }
+        return testSteps;
```

Comment:
seems like a circular execution, can we have proper doc over what is been processed over here?